language/bytecode_verifier/src/struct_defs.rs

```diff
@@ -97,7 +97,7 @@ impl<'a> StructDefGraphBuilder<'a> {
     ) -> impl Iterator<Item = StructDefinitionIndex> + 'a {
         let struct_def = self.module.struct_def_at(idx);
         let struct_def = StructDefinitionView::new(self.module, struct_def);
-        let fields = struct_def.fields();
+        let fields = struct_def.fields().into_iter().flatten();
```

Comment:
It's a bit hard to tell this is dealing with the `Native` case. A comment or an explicit pattern match might help?